### DISCUSSION TITLE
Suggestion: Motion Guard link to local recording

### DISCUSSION THREAD

**Discussion by ScottInBristol** [themactep/thingino-firmware#289] [Ideas]
Hi,
As well as Motion Guard detection leading to sending a snapshot to one of the selected destination services, it would be very useful to have an option to save X seconds of video as a local recording. I'm sure individual users could adapt the code to make this happen, but for those of us without the skillset to do it ourselves, any chance this could be added to a future update? 
Thanks!

**Comment by spikenipple** [comment]
This is a great idea - this or a way to delay the snapshot taken by motion detection by a specified amount. My cameras detect motion but the snaps show nothing of note as it snaps the very instant that motion is detected.
Thanks

**Comment by smshr** [comment]
Great Idea. It would be even better if Motion Guard can store few seconds before & after motion detection (say 10s). RAM can be used as a loop buffer to store first 10s (or less depending on the RAM size) of video which gets dumped to local SD Card (or ftp) once object gets detected. This will also prevent SD card wear and tear.

**Comment by Jorgi-1** [comment]
This was enquired about recently in discord and it looks like there are no plans to support on camera recording on motion detected, which is a shame as I think this feature could work really nice paired with continuous recording and help retain the features of stock camera firmware without the need to use stock cameras software.

**Reply by gtxaspec** [reply]
feature is planned. you'll know when its available. =D

**Reply by Jorgi-1** [reply]
> feature is planned. you'll know when its available. =D

My apologies, I was just relaying the information I found searching for the answer in discord. Will look forward to the feature, thanks you guys for supporting all these great projects.

**Reply by drew5494** [reply]
Hello was this ever implemented? I would like this feature too.  @gtxaspec 
A motion notification saving to the device instead of being sent via the web.

**Comment by drew5494** [comment]
Created pull request https://github.com/themactep/thingino-firmware/pull/681

**Reply by virmaior** [reply]
I like the idea but the vbuffer code depends on openRTSP's ability to capture well. Which in my experience it produces amazingly glitchy video. Maybe that's good enough for this purpose?

**Reply by drew5494** [reply]
yes, even a simple image capture is good enough for me. just something to let me know at what time there was motion.  @virmaior

**Comment by karlitos** [comment]
This is very interesting suggestion. I tried the [Yi Hack for the Yi cameras](https://github.com/roleoroleo/yi-hack-Allwinner-v2) and having local recordings on a SD card when motion was detected is the only feature I am missing so far in Thingino. In all other regards Thingino feels way more mature.

The Yi Hack allows to view and delete the recordings in the Web UI
![Example of the Yi Hack UI](https://user-images.githubusercontent.com/63554131/125061713-e8861f00-e0ad-11eb-98e2-bc5537ba21bd.png)
If the local recording would be implemented I'd be glad to help with implementing the Video-Player  and some simple recording-management (delete, share ...) in the Web UI

**Comment by gupta-ash** [comment]
It would be great to have the flexibility to place the local recording on the mounted NFS share.

**Reply by virmaior** [reply]
I think that's probably not a great idea.
I was doing local recordings as I was when I was using wyze/atom stock firmwares, but the recording quality on the local camera is quite low because openRTSP is not actually that good at consuming the stream. It produces large numbers of key-frame related artifacts. 

Conversely, using ffmpeg remotely produces clean streams so if possible set the NFS-running computer to consume the streams. (if unedited, this takes up little processing power -- like 1% per stream on an RPI 4).

**Reply by drew5494** [reply]
on my Eufy E220 it seems to look fine with telegram notifications.  Some frames might be dropped but nothing terrible.

**Comment by Wex005** [comment]
Local motion recording would be a great idea!!!

I am setting up a Wyze Cam v2 with Thingino firmware in a vacant house. There have been items stored in the house, walk off. The house has no electricity or Internet access. The camera is powered by a 30,000 mah battery bank. I am needing to record motion to the local SD Card.

Can the Timelaspe Recorder and/or Video Recorder methods be used as a Motion Guard notification method?
